ts/src/profiler.ts

```diff
@@ -81,16 +81,34 @@ function getServerResponseBackoff(response: http.IncomingMessage): number|
     undefined {
   // tslint:disable-next-line: no-any
   const body = (response as any).body;
-  if (body && body.error && body.error.details &&
-      Array.isArray(body.error.details)) {
-    for (const item of body.error.details) {
-      if (typeof item === 'object' && item.retryDelay &&
-          typeof item.retryDelay === 'string') {
-        const backoffMillis = parseDuration(item.retryDelay);
-        if (backoffMillis > 0) {
-          return backoffMillis;
-        }
-      }
+
+  // The response currently does not have field containing the server-specified
+  // backoff. As a workaround, response body's message is parsed to get the
+  // backoff.
+  // TODO (issue #250): Remove this workaround and get the retry delay from
+  // body.error.details.
+  if (body && body.message && typeof body.message === 'string') {
+    return parseBackoffDuration(body.message);
+  }
+  return undefined;
+}
+
+
+const BACKOFF_MSG_PATH =
```

Comment:
Why is it "PATH"? Did you mean "_PAT" for "pattern"? Maybe use "_REGEX", I think it would be a bit cleaner.

Reply:
Done.